core/frontend/helpers/ghost_foot.js

```diff
@@ -2,24 +2,31 @@
 // Usage: `{{ghost_foot}}`
 //
 // Outputs scripts and other assets at the bottom of a Ghost theme
-var proxy = require('./proxy'),
-    _ = require('lodash'),
-    SafeString = proxy.SafeString,
-    settingsCache = proxy.settingsCache;
+const proxy = require('./proxy');
+const _ = require('lodash');
+const SafeString = proxy.SafeString;
+const api = proxy.api;
 
 // We use the name ghost_foot to match the helper for consistency:
 module.exports = function ghost_foot(options) { // eslint-disable-line camelcase
-    var foot = [],
-        globalCodeinjection = settingsCache.get('ghost_foot'),
-        postCodeinjection = options.data.root && options.data.root.post ? options.data.root.post.codeinjection_foot : null;
-
-    if (!_.isEmpty(globalCodeinjection)) {
-        foot.push(globalCodeinjection);
-    }
-
-    if (!_.isEmpty(postCodeinjection)) {
-        foot.push(postCodeinjection);
-    }
-
-    return new SafeString(foot.join(' ').trim());
+    const apiVersion = data.root._locals.apiVersion;
+    let settingsController = api[apiVersion].publicSettings || api[apiVersion].settings;
```

Comment:
Think this piece belongs conceptually somewhere else, for example in proxy. Becaue on the helper level, we ideally only refer to a specific controller that proxy provided us with without choosing apiVerison. 

Maybe a good way to start with this would be assigning correct `settings` controller inside of `proxy.api` object? This would also move into direction already mentioned int the comment there: `// TODO: Expose less of the API to make this safe`

So in `psoxy.js` exposed api property could look something like (warning! almost pseudocode):
```
api: {
	...require('../../server/api'),
	settings: api[apiVersion].publicSettings || api[apiVersion].settings
},

Reply:
@gargol We currently use this pattern in some helpers like [here](https://github.com/TryGhost/Ghost/blob/master/core/frontend/helpers/prev_next.js#L57), but I agree with the overall direction above 💯, and ideally this should be combined with solution for issue [here](https://github.com/TryGhost/Ghost/issues/10548). One challenge here is the `apiVersion` value has to be passed through helper as `proxy` won't have the value of `apiVersion` and can only come in via parameters passed to helpers.

A possible solution to bypass it could be a `getAPI` helper util method which takes in `apiVersion` and uses [route config objects](https://github.com/TryGhost/Ghost/issues/10548) to return the right API, which then is used from all helpers like `prev_next` etc to fetch the API. Wdyt?

Reply:
Yeah, creating a helper that would encapsulate controller logic would be a good approach as well :+1:  The direction, I think, should be towards having a single place where we can define and limit communication between helpers and the API.

There are multiple places in helpers that have code of this kind `controller = api[apiVersion]`. The most advanced use of this syntax is in `get` helper as it accesses a few resources in a generic way. Because 'setting' is also a resource available through Content API, don't see a reason why not treat it in a similar way :thinking: 	

And once such refactor is done we can maybe get to some nicer syntax like: `apiHelper.setting({apiVersion}).browse()` or something similar in case of get/prev_next helpers.